src/main/java/com/example/Ticket.java

```diff
@@ -1,25 +1,44 @@
 package com.example;
 
+import java.sql.Timestamp;
 import java.time.Instant;
 import java.util.Objects;
+import jakarta.persistence.*;
 
+@Entity
+@Table(name = "ticket")
 public class Ticket extends Identifiable {
 
     @NullableWarning
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+    @Column(name = "creation_date")
+    private Timestamp creationDate;
+    @Column(name = "ticket_type", nullable = false)
+    @Enumerated(EnumType.STRING)
+    private TicketType ticketType;
+    @Column(name = "user_id")
+    private int userId;
+
+    @Transient
+    private long time;
+    @Transient
     private String concertHall;
+    @Transient
     private String eventCode;
-    private long time;
+    @Transient
     private boolean isPromo;
+    @Transient
     private char stadiumSector;
+    @Transient
     private double maxBackpackWeight;
+    @Transient
     private double price;
-    private final long creationTime = Instant.now().toEpochMilli();  // Creation time set at object instantiation
-    private TicketType ticketType; // Add this line
-    private int userId;
-    private int id;  // Auto-generated by DB
-    private long creationDate;
+
 
     public Ticket() {
+        this.creationDate = new Timestamp(System.currentTimeMillis());
         NullableWarningChecker.check(this);
```

Comment:
after NullableWarningChecker.check(this);

Reply:
I mean - first validation, if all OK, than this.creationDate = new Timestamp(System.currentTimeMillis());